File: compiler/testData/compileKotlinAgainstJava/Inheritance.java
package test;

public class Inheritance extends AbstractInheritance { }

abstract class AbstractInheritance implements Interface {

    @Override public int getAnswer() { return 42; }

}

interface Interface extends I {
    @Override
    int getAnswer();
}

interface I {
    int getAnswer();
}

interface I2 extends I {

    @Override
    int getAnswer();

}

File: plugins/kapt-lite/kapt-lite-compiler-plugin/testData/stubs/inheritance.kt
package inheritance

interface Intf
abstract class Base

class Foo : Base(), Intf

//

interface A1 {
    fun foo()
}

interface A2 {
    fun bar()
}

class A1A2 : A1, A2 {
    override fun foo() {}
    override fun bar() {}
}

//

interface B1 {
    fun foo() {}
}

interface B2 {
    fun foo() {}
}

class B1B2 : B1, B2 {
    override fun foo() {}
}

//

abstract class C1
abstract class C2 : C1()
abstract class C3 : C1()
class C4 : C3(), Intf, A1, A2, B1, B2 {
    override fun foo() {}
    override fun bar() {}
}